### ISSUE TITLE
Changes to key vault configuration are not applied if a key vault is recovered during creation

### ISSUE THREAD

**Issue by culpinnis** [hashicorp/terraform-provider-azurerm#17375]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.3

### AzureRM Provider Version

3.8.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 1.0.0"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = ">=3.8.0"
    }
  }
}
provider "azurerm" {
  features {
    key_vault {
      purge_soft_delete_on_destroy    = false
      recover_soft_deleted_key_vaults = true
    }
  }
  subscription_id = var.target_subscription
}

### Locals
locals {
}

### Variables
variable "vault_name" {
  description = "Name of the key vault that is created"
  type        = string
}
variable "target_subscription" {
  description = "The subscription id"
  type        = string
}
variable "default_location" {
  description = "The default Azure region"
  type        = string
  default     = "Germany West Central"
}
variable "target_resource_group" {
  description = "The resource group for deploying the vault"
  type        = string
}

### Data
data "azurerm_client_config" "current" {
}

data "azurerm_resource_group" "target_resource_group" {
  name = var.target_resource_group
}

### Key vault
resource "azurerm_key_vault" "vault" {
  name                        = var.vault_name
  location                    = var.default_location
  resource_group_name         = data.azurerm_resource_group.target_resource_group.name
  enabled_for_disk_encryption = false
  tenant_id                   = data.azurerm_client_config.current.tenant_id
  soft_delete_retention_days  = 7
  purge_protection_enabled    = false
  sku_name                    = "standard"

  network_acls {
    default_action = "Deny"
    bypass         = "AzureServices"
  }
  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    secret_permissions = [
      "Delete", "Get", "List", "Set"
    ]

    certificate_permissions = [
      "Get", "Update", "Import", "Delete", "Purge"
    ]
  }

  lifecycle {
    ignore_changes = [
      access_policy,
      tags,
    ]
  }
}
```

### Debug Output/Panic Output

```shell
https://futureconsultant-my.sharepoint.com/:u:/g/personal/culpinnis_thinkport_digital/EUGEngEW_zRAlae0V9rmqEMB0csTtDlKEhCcYBp2beSeyA?e=uRoyoW

apply_1 -> Initial creation
destroy -> Destruction after setting bypass to "No" in the portal
apply_2 -> Recreation of the key vault
```

### Expected Behaviour

When applying a TF configuration the created resource should have the same configuration as in the state file. 

### Actual Behaviour

The first creation of the key vault is always successful, but we encountered problems when the key vault is soft-deleted and recovered: If you change your TF arguments (e.g. change bypass from "none" to "AzureServices") terraform will recover the key vault, but it will not apply your changes. The created key vault will have the same configuration as the deleted one. Running terraform a second time will fix the configuration. 

Another scenario is a modification outside of TF: When the created key vault was modified outside of TF and deleted afterwards, the newly created/recovered key vault will have the same settings as the deleted one. 

### Steps to Reproduce

2. Navigate to your key vault in the portal, select Networking and set "Allow trusted Microsoft services to bypass this firewall" to "No"
3. terraform destroy
4. terraform apply
5. Check the key vault in the portal: The option will be still set to "No" even if it is set the AzureServies in the TF code

OR:

1. terraform apply
2. terraform delete
3. Modify TF bypass code to "None"
4. terraform apply
5. Check the key vault in the portal: The option will be set to "Yes" instead of "No"

### Important Factoids

Running in Germany

**Comment by sinbai** [comment]
@culpinnis thank you for opening this issue here. Seems this is an azure API issue since in the API request sent by terraform bypass has been set to `AzureServies` in  step 4, but azure API return None. [An azure API issue](https://github.com/Azure/azure-rest-api-specs/issues/19606) has been filed. Could you please follow it for more updates?

**Comment by ramonschopper** [comment]
@sinbai I can confirm this issue but not only for `bypass` but all networking configuration under the `network_acls` block like `ip_rules` or `default_action`.

**Comment by prashantguleria** [comment]
> @sinbai I can confirm this issue but not only for `bypass` but all networking configuration under the `network_acls` block like `ip_rules` or `default_action`.

Yes, Even I am facing it, If I re-create the key vault with same name it is not updating the netwrok acls.

**Comment by kemaldev** [comment]
I can also confirm that this issue still exists when trying to create a KeyVault with different ip_rules set after destroying it. The ip_rules will not be applied on the KeyVault. It will keep the old ip_rules that were set before deletion.

**Comment by JeremyMCastillo** [comment]
I am also currently having this issue using Terraform `1.9.8` and azurerm provider `4.5.0`. The `ip_rules` from a previously deployed key vault are recreated even though the configuration has been updated.

**Comment by whatisthejava** [comment]
A manual workaround is to delete all the IP rules and set public network on, this does allow teraform to rebuild the IP list correctly.
